presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java

```diff
@@ -159,6 +172,9 @@ public QueryObjectBundle rewriteQuery(@Language("SQL") String query, ClusterType
             ResultSetMetaData metadata = getResultMetadata((Query) statement);
             List<Identifier> columnAliases = generateStorageColumnAliases(metadata);
             Query rewrite = rewriteNonStorableColumns((Query) statement, metadata);
+            if (functionCallRewriter.isPresent()) {
```

Comment:
Why is the function call rewrite only done for select queries ?

Reply:
I'm curious is where does it say it is only for "select" queries?

Reply:
The above two if clauses are for insert and ctas -- so this one is for select queries. We should be rewriting the above two as well I think. Esp for batch workload which has a lot of inserts.

Reply:
The if-flow here now is
```
if (statement instanceof CreateTableAsSelect) {
}
if (statement instanceof Insert) {
}
if (statement instanceof Query) {
    rewrite = (Query) functionCallRewriter.get().process(rewrite);
}
if (statement instanceof CreateView) {
}
```

I will add `functionCallRewriter.get().process()` to the rest in my 2nd PR, https://github.com/prestodb/presto/pull/18095. This PR adds it to the Query branch, only for the sake of unit tests that utilizes existing setup.

This change is split into 2 PRs,
This https://github.com/prestodb/presto/pull/20923 to add the FunctionCallRewriter class.
https://github.com/prestodb/presto/pull/18095 to invoke the FunctionCallRewriter class in the QueryRewriter class  as well as configs.